browser/src/control/jsdialog/Widget.TreeView.ts

```diff
@@ -521,6 +532,78 @@ class TreeViewControl {
 		}
 	}
 
+	createQuickFindResult(
+		parent: HTMLElement,
+		entry: TreeEntryJSON,
+		index: any,
+		builder: JSBuilder,
+		searchTerm: string,
+	) {
+		const text = entry.text;
+		if (this.isPageDivider(text)) {
+			const cell = L.DomUtil.create(
+				'span',
+				builder.options.cssClass +
+					` ui-treeview-cell-text ui-treeview-cell-text-content ui-treeview-${entry.row}-${index} page-divider`,
+				parent,
+			);
+			cell.innerText = text.substring(1); // Remove leading '-'
+			return cell;
+		}
+
+		const searchPattern = `[${searchTerm}]`;
```

Comment:
what about using the searchPattern as a "delimiter" ? No manual index calculation this way:

```
const fragments = sourceText.split(searchPattern);
if (fragments.length === 1) {
    /// not found
    mainSpan.appendChild(document.createTextNode(sourceText));
} else {
    // found, can be many times
    for (const i = 0; i < fragments.length - 1; i++) {
        mainSpan.appendChild(document.createTextNode(fragments[i]); // pre
        mainSpan.appendChild(   .. here highlighted searchTerm ... );
    }
    mainSpan.appendChild(document.createTextNode(fragments[fragments.length - 1]); // post
}
```